pkg/secretless/entrypoint/entrypoint.go

```diff
@@ -50,49 +53,88 @@ func StartSecretless(params *SecretlessOptions) {
 		log.Fatalln(err)
 	}
 
-	// Exit signal listener
-
-	// Coordinates processes interested in exit signals
-	exitListener := signal.NewExitListener()
-
 	// Optional Performance Profiling
-
 	handlePerformanceProfiling(params.ProfilingMode, exitListener)
 
-	// Start Services
-
-	allServices := proxyservice.NewProxyServices(cfg, availPlugins, logger, evtNotifier)
-	exitListener.AddHandler(func() {
-		fmt.Println("wait for all services signal")
-		err := allServices.Stop()
-		if err != nil {
-			// Log but but allow cleanup of other subscribers to continue.
-			log.Println(err)
-		}
-	})
+	configChangedChan, err := newConfigChangeChan(
+		params.ConfigFile,
+		params.ConfigManagerSpec,
+		params.FsWatchEnabled,
+	)
+	if err != nil {
+		log.Fatalln(err)
+	}
```

Comment:
Put the health check stuff 👇 above this channel block ☝️ .  

Otherwise it's like we're starting on the topic of config reloading, randomly mentioning health checks, then going back to talking about config reloading. 

Also add section comments (comments with blank line above and below) for each of the config pieces:

```
// Create channel new configs

// Create config reload function 
//   --> this can become a single line with the function pulled out

// etc
```